### DISCUSSION TITLE
How to disable filtering?

### DISCUSSION THREAD

**Discussion by crates51** [ibhagwan/fzf-lua#1460] [Q&A] [answered]
Hello

Is there a easy way to disable filtering and just navigate trough results using j (down) k(up). unlike the default way of ctrl+j, ctrl+k ?
(Like a quick fix list) Asking for diagnostics, i dont really need to search in them.

I found this:

        -- set `ignore_patterns = false` to disable filtering

But not sure where to put it, or if that is what i am looking for

Also, it would be cool later on, if i really need filtering, to activate it somehow.. when inside diagnostics panel

Thaanks

**Comment by ibhagwan** [comment]
Once the results are displayed you’re in the realm of fzf the binary, read `man fzf` to see what’s possible, if you wanna set new keymaps for `j|k` use `keymap.fzf` table, for flags use `fzf_opts` table:
```lua
lua require("fzf-lua").files { keymap={fzf={j="down",k="up"}} }
```
> Also, it would be cool later on, if i really need filtering, to activate it somehow.. when inside diagnostics panel

You can easily do that with a keymap that unbinds `j|k`, for example with `q`:
```lua
lua require("fzf-lua").files { keymap={fzf={j="down",k="up",q="unbind(j)+unbind(k)"}} }
```

You can have another key to `rebind` jk if you wish, `man fzf` is your friend.

**Comment by crates51** [comment]
Ok, so i found somethin:

"To prevent fzf from doing the filtering, we use `--disabled` option. Now fzf is a mere selector interface. "

So this works for me:

    keymap.set("n", "<leader>d", function()
      fzflua.diagnostics_document({
        winopts = {
          height = 0.4,
          width = 1.0,
          row = 1.0,
          preview = {
            vertical = "up:70%",
          },
        },
        previewer = "builtin", -- diagnostic details preview
        ignore_patterns = false,
        fzf_opts = {
          ["--layout"] = "reverse-list",
          ["--disabled"] = "",
        },
        actions = {
          ["default"] = actions.lsp_definitions, -- action when selecting a diagnostic
        },
        keymap = {
          fzf = {
            ["j"] = "down",
            ["k"] = "up",
          },
        },
      })
    end, { desc = "Show LSP Diagnostics for Current File" })

But there would be one last thing to be the selector i want.. is that input line, can i get rid of it?

![image](https://github.com/user-attachments/assets/8166722b-86c2-4b39-8911-611ca3c9a8da)

**Comment by ibhagwan** [comment]
> But there would be one last thing to be the selector i want.. is that input line, can i get rid of it?

You can set it to an empty prompt but I’m not sure if you can get rid of it completely.

Perhaps you can set it to empty or to `———-` and use `--border` to remove the bottom border?

**Reply by crates51** [reply]
Yeah.. but even if change the prompt, entering letter like "a" "b" "c".. etc will still apear there and look wierd, neverthless i found this: https://github.com/junegunn/fzf/issues/2890

Look something similar to my problem:

You can set `--bind='change:clear-query' `to not let to appear the letter you enter 
 and also clean the prompt with `--prompt "" ` and `--no-info` to get rid of additional infos near the prompt
 
 So a final fzf command would look like:  `fzf --disabled --bind='change:clear-query' --no-info --prompt ""
` 

So i tried:

```
        fzf_opts = {
          ["--layout"] = "reverse-list",
          ["--disabled"] = "",
          ["--bind"] = "change:clear-query",
        },
```
But the letters still appear as i type

In fzf it works, i tried

Any idea why `change:clear-query` doesnt work in fzf-lua?

**Comment by ibhagwan** [comment]
> Any idea why change:clear-query doesnt work in fzf-lua?

No idea, the flag should be proxies as is, add `debug=true` and you’ll be able to see the generated command in `:messages`.

**Comment by crates51** [comment, marked as answer]
Ok, i finally (almost) achieved what i wanted.

I had to use fzf_args to send the arguments Raw like so

`        fzf_args = "--disabled --bind='change:clear-query' --no-info --prompt ''",
`
```
    keymap.set("n", "<leader>d", function()
      fzflua.diagnostics_document({
        winopts = {
          height = 0.4,
          width = 1.0,
          row = 1.0,
          preview = {
            vertical = "up:70%",
          },
        },
        previewer = "builtin", -- diagnostic details preview
        ignore_patterns = false,
        fzf_opts = {
          ["--layout"] = "reverse-list",
        },
        fzf_args = "--disabled --bind='change:clear-query' --no-info --prompt ''",
        actions = {
          ["default"] = actions.lsp_definitions, -- action when selecting a diagnostic
        },
        keymap = {
          fzf = {
            ["j"] = "down",
            ["k"] = "up",
          },
        },
      })
    end, { desc = "Show LSP Diagnostics for Current File" })

```
And it looks pretty good as a selector:

![image](https://github.com/user-attachments/assets/2c041ebe-db6d-4908-b592-411bc83c6407)

-no-info was nice to get rid of those info in the bottom right corner

![image](https://github.com/user-attachments/assets/35302de9-5faa-42cc-9c1d-44050964db67)


I mean, is like 99% perfect, the only little inconvenience would be the cursor, but eh, you can't have them all, i will mark this as resolved, thanks !!

**Comment by ibhagwan** [comment]
Btw, the reason adding `--bind` in `fzf_opts` didn't work is because the bind string is crafted from `keymap.fzf`, although purely cosmetic I would write it like this:
```lua
  prompt = "", -- instead of `--prompt`
  fzf_opts = {
    ["--layout"] = "reverse-list",
    ["--disabled"] = true,
    ["--no-info"] = true,
  },
  keymap = {
    fzf = {
      ["j"] = "down",
      ["k"] = "up",
      ["change"] = "clear-query", -- instead of using `fzf_opts.--bind`
    },
  },
```
